src/app_service/service/community/service.nim

```diff
@@ -197,8 +200,16 @@ QtObject:
           community.pendingRequestsToJoin.add(membershipRequest)
           self.joinedCommunities[membershipRequest.communityId] = community
           self.events.emit(SIGNAL_COMMUNITY_EDITED, CommunityArgs(community: community))
-
           self.events.emit(SIGNAL_NEW_REQUEST_TO_JOIN_COMMUNITY, CommunityRequestArgs(communityRequest: membershipRequest))
+      
+      if(receivedData.declinedMembershipRequests.len > 0):
+        for membershipRequest in receivedData.declinedMembershipRequests:
+          # remove my pending requests
+          keepIf(self.myCommunityRequests, request => request.communityId != membershipRequest.communityId)
+          self.myCommunityRequests.add(membershipRequest)
+          self.events.emit(SIGNAL_COMMUNITY_MY_REQUEST_REJECTED, CommunityRequestArgs(communityRequest: membershipRequest))
+          singletonInstance.globalEvents.myRequestToJoinCommunityRejected("Community Request Rejected",
+          fmt "Your request to join community is rejected", membershipRequest.communityId)
```

Comment:
nit: can you add an indent on those two lines to show they are the same as the event

Reply:
fixed